lib/LogicalExpression.h

```diff
@@ -510,37 +505,34 @@ namespace LogicalExpressionDetail
 template<typename ContainedClass>
 class LogicalExpression
 {
-	typedef LogicalExpressionDetail::ExpressionBase<ContainedClass> Base;
-public:
-	/// Type of values used in expressions, same as ContainedClass
-	typedef typename Base::Value Value;
-	/// Operators for use in expressions, all include vectors with operands
-	typedef typename Base::OperatorAny OperatorAny;
-	typedef typename Base::OperatorAll OperatorAll;
-	typedef typename Base::OperatorNone OperatorNone;
-	/// one expression entry
-	typedef typename Base::Variant Variant;
-
-private:
-	Variant data;
-
-public:
-	/// Base constructor
-	LogicalExpression()
-	{}
-
-	/// Constructor from variant or (implicitly) from Operator* types
-	LogicalExpression(const Variant & data):
-		data(data)
-	{
-	}
+	using Base = LogicalExpressionDetail::ExpressionBase<ContainedClass>;
```

Comment:
indentation still dead

Reply:
@IvanSavenko where? I do not see any indentation errors AFAIK. Or you mean Variant (I need to split it into multiple lines)?

Reply:
Lines 510-535 here?

Reply:
Move it 1 level backwards?

Reply:
Yes. Class is defined at top-level, at 0 indentation level. All members should be placed at 1 tab, not 2

Reply:
done